R/utils.R

```diff
@@ -162,10 +164,24 @@ create_placeholder = function(x) {
   h1 = grep(reg_part, h, value = TRUE)  # part title
   h2 = grep(reg_app, h, value = TRUE)   # appendix title
   h3 = setdiff(h, c(h1, h2))
-  h4 = grep('^#{2,} ', x, value = TRUE)  # section/subsection/... titles
+
+  poss_sections = grep('^#{2,} ', x)  # section/subsection/... titles
+  code_blocks = grep('^```', x)
+
+  # Checks for ^#{2,} potentially occuring in a code block
+  # for each match count the preceeding occurances of ^```
+  #  - odd counts = in code block
+  #  - even counts = section / subsection / etc.
+  in_block = vapply(poss_sections, function(x) as.logical(sum(x>code_blocks) %% 2), TRUE)
+
+  sections = poss_sections[!in_block]
+  h4 = x[sections]
```

Comment:
I don't have a strong opinion on this. I'm not against your improvement, but I don't mind if certain comments in the R code are treated as section headings, either. As I said, it is for preview only, and rough edges should be fine.

Reply:
@rundel FYI: With `devtools::install_github(c('yihui/xaringan', 'rstudio/bookdown'))`, comments in fenced code blocks will no longer be treated as section headers.

Reply:
👍